store/outputStore.go

```diff
@@ -0,0 +1,375 @@
+/* Getters */
+
+// GetAccount returns the account at the associated address
+func (store OutputStore) GetAccount(ctx sdk.Context, addr common.Address) (Account, bool) {
+	key := prefixKey(accountKey, addr.Bytes())
+	data := store.Get(ctx, key)
+	if data == nil {
+		return Account{}, false
+	}
+
+	var acc Account
+	if err := rlp.DecodeBytes(data, &acc); err != nil {
+		panic(fmt.Sprintf("transaction store corrupted: %s", err))
+	}
+
+	return acc, true
+}
+
+// GetDeposit returns the deposit at the given nonce
+func (store OutputStore) GetDeposit(ctx sdk.Context, nonce *big.Int) (Deposit, bool) {
+	key := prefixKey(depositKey, nonce.Bytes())
+	data := store.Get(ctx, key)
+	if data == nil {
+		return Deposit{}, false
+	}
+
+	var deposit Deposit
+	if err := rlp.DecodeBytes(data, &deposit); err != nil {
+		panic(fmt.Sprintf("deposit store corrupted: %s", err))
+	}
+
+	return deposit, true
+}
+
+// GetFee returns the fee at the given position
+func (store OutputStore) GetFee(ctx sdk.Context, pos plasma.Position) (Output, bool) {
+	key := prefixKey(feeKey, pos.Bytes())
+	data := store.Get(ctx, key)
+	if data == nil {
+		return Output{}, false
+	}
+
+	var fee Output
+	if err := rlp.DecodeBytes(data, &fee); err != nil {
+		panic(fmt.Sprintf("output store corrupted: %s", err))
+	}
+
+	return fee, true
+}
+
+// GetOutput returns the output at the given position
+func (store OutputStore) GetOutput(ctx sdk.Context, pos plasma.Position) (Output, bool) {
+	// allow deposits to returned as an output
+	if pos.IsDeposit() {
+		return store.depositToOutput(ctx, pos.DepositNonce)
+	}
+
+	key := prefixKey(positionKey, pos.Bytes())
+	hash := store.Get(ctx, key)
+
+	tx, ok := store.GetTx(ctx, hash)
+	if !ok {
+		return Output{}, ok
+	}
+
+	output := Output{
+		Output:  tx.Transaction.Outputs[pos.OutputIndex],
+		Spent:   tx.Spent[pos.OutputIndex],
+		Spender: tx.Spenders[pos.OutputIndex],
+	}
+
+	return output, ok
+}
+
+// GetTx returns the transaction with the provided transaction hash
+func (store OutputStore) GetTx(ctx sdk.Context, hash []byte) (Transaction, bool) {
+	key := prefixKey(hashKey, hash)
+	data := store.Get(ctx, key)
+	if data == nil {
+		return Transaction{}, false
+	}
+
+	var tx Transaction
+	if err := rlp.DecodeBytes(data, &tx); err != nil {
+		panic(fmt.Sprintf("transaction store corrupted: %s", err))
+	}
+
+	return tx, true
+}
+
+// GetTxWithPosition returns the transaction that contains the provided position as an output
+func (store OutputStore) GetTxWithPosition(ctx sdk.Context, pos plasma.Position) (Transaction, bool) {
+	key := prefixKey(positionKey, pos.Bytes())
+	hash := store.Get(ctx, key)
+	return store.GetTx(ctx, hash)
+}
+
+// -----------------------------------------------------------------------------
+/* Has */
+
+// HasAccount returns whether an account at the given address exists
+func (store OutputStore) HasAccount(ctx sdk.Context, addr common.Address) bool {
```

Comment:
Maybe using `Account` here is not the best since this still isn't an Account model. Maybe have something to be misleading, `UTXOAccount` or `UTXOHolder` something like that

Reply:
I think its fine to abstract away the UTXO model in the same way Coinbase and other wallets do. Maybe `Wallet` would be more appropriate?

Reply:
I like `Wallet`, I don't want to use terminology that is already established to mean something different.